tests/test_registers.py

```diff
@@ -0,0 +1,87 @@
+# Copyright 2018-2024 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Unit tests for pennylane.registers.
+"""
+
+import pytest
+
+import pennylane as qml
+from pennylane.wires import Wires
+
+
+class TestRegisters:
+    @pytest.mark.parametrize(
+        "wire_dict, expected_register",
+        [
+            (
+                {"alice": 3, "bob": {"nest1": 3, "nest2": 3}, "cleo": 3},
+                {
+                    "alice": Wires([0, 1, 2]),
+                    "nest1": Wires([3, 4, 5]),
+                    "nest2": Wires([6, 7, 8]),
+                    "bob": Wires([3, 4, 5, 6, 7, 8]),
+                    "cleo": Wires([9, 10, 11]),
+                },
+            ),
+            (
+                {"alice": 3, "bob": {"nest1": 3}, "cleo": 3},
+                {
+                    "alice": Wires([0, 1, 2]),
+                    "nest1": Wires([3, 4, 5]),
+                    "bob": Wires([3, 4, 5]),
+                    "cleo": Wires([6, 7, 8]),
+                },
+            ),
+            (
+                {"alice": 3, "bob": {"nest1": {"nest2": {"nest3": 1}}}, "cleo": 3},
+                {
+                    "alice": Wires([0, 1, 2]),
+                    "nest3": Wires([3]),
```

Comment:
I see you're doing a DFS traversal for the construction over the nestedness. So, I think it would be wise to leave a note on the construction order of the register dictionary.